### DISCUSSION TITLE
Cross-filtering multiple datasets

### DISCUSSION THREAD

**Discussion by marksmall** [visgl/loaders.gl#2236] [Q&A]
Hi there, I'm asking here to get some advice on how to approach a problem, I have some ideas, but would like the benefit of the communities collective experience.

### Problem

I have multiple datasets, that we want to cross-filter on. Say MVT data on age and another MVT dataset on fuel poverty. We want to be able to load the data for one, then filter that dataset, say the age, by those under 15. Then use that filtered data, to then filter another dataset by the level of fuel poverty, so we can say, the number of 15 yr olds and below in poverty are X, but the number of 60 yr olds and above are in poverty Y.

I understand how to do this if it were 1 single dataset, but not sure where to start when trying to use 2 separate datasets. I've had a couple of ideas of where to start e.g.

- extend the `CompositeLayer`, to have a limit of 4 MVTLayers, each with their own dataset and somehow switch between each to set/update a filter on that data, which would in turn update the next layer with it's new value, to re-evaluate that filter and so on.
- extend the MVT loader, so as to load multiple datasets instead of just one and maybe combine that data into one, that would be consumed by a regular MVTLayer and filter it as if it were one dataset.

Of course I've no idea yet if either are even possible, I'm just at the investigation phase of this work and I'm canvasing for advice. Thanks in advance.

**Comment by ibgreen** [comment]
I think that  "multi-tileset loader" is an interesting idea. When doing analytics on multiple tiled datasets, it could be interesting to have a tile loader that handles multiple tilesets. 

It would allow the app to get a list of only those tiles for which all tilesets have loaded, and it could also prioritize loading of tiles so that the same tiles for each dataset are loaded first, so that analytics can begin early.

I half started on an RFC for this at one point, but it would require more work than I could justify investing at the time.

Implementation-wise, it is possible that one could build this as an adapter or orchestrator for the Tileset2D class. Perhaps an "unoptimized" version would not require too much effort.

I think such a class could go into loaders.gl. But then, I also happen to think the Tileset2D class should be located in loaders.gl, not in deck.gl.
